master/internal/api_experiment.go

```diff
@@ -230,16 +230,17 @@ func (a *apiServer) deleteExperiment(exp *model.Experiment, user *model.User) er
 	if err != nil {
 		return err
 	}
-
-	addr := actor.Addr(fmt.Sprintf("delete-checkpoint-gc-%s", uuid.New().String()))
-	jobSubmissionTime := exp.StartTime
-	taskID := model.NewTaskID()
-	ckptGCTask := newCheckpointGCTask(
-		a.m.rm, a.m.db, a.m.taskLogger, taskID, exp.JobID, jobSubmissionTime, taskSpec, exp.ID,
-		conf, checkpoints, true, agentUserGroup, user, nil,
-	)
-	if gcErr := a.m.system.MustActorOf(addr, ckptGCTask).AwaitTermination(); gcErr != nil {
-		return errors.Wrapf(gcErr, "failed to gc checkpoints for experiment")
+	if len(checkpoints) > 0 {
+		addr := actor.Addr(fmt.Sprintf("delete-checkpoint-gc-%s", uuid.New().String()))
+		jobSubmissionTime := exp.StartTime
+		taskID := model.NewTaskID()
+		ckptGCTask := newCheckpointGCTask(
```

Comment:
it's fine to me if it is duplicated, but we should push this into `newCheckpointGCTask` (i guess the actor's prestart) otherwise this is just gonna happen again.

Reply:
Something like this (in checkpoint_gc.go):

```
	case actor.PreStart:
		if t.ToDelete == "" {
			t.completeTask(ctx)
			return nil
		}
```

Or is there more to it?

Reply:
ahh... deleting tensorboards is very good catch. we can never know upfront if there is tb data to gc without checking in the gc task.. i think we may need to modify the gc task itself/how it is launched to be ok with `--delete []` or something.